src/inline.tsx

```diff
@@ -5,30 +5,37 @@ import {getPrefixedDataAttributes} from './utils/dom';
 import * as styles from './inline.css';
 import {applyCssVars} from './utils/css';
 
-import type {DataAttributes} from './utils/types';
+import type {ByBreakpoint, DataAttributes} from './utils/types';
+
+type NumericSpace = -16 | -12 | -8 | -4 | -2 | 0 | 2 | 4 | 8 | 12 | 16 | 24 | 32 | 40 | 48 | 56 | 64;
+type FlexSpace = 'between' | 'around' | 'evenly';
+
+const calcSpaceValue = (space: NumericSpace | FlexSpace) => {
+    if (typeof space === 'number') {
+        return `${space}px`;
+    } else {
+        return `space-${space}`;
+    }
+};
+
+const calcInlineVars = (space: FlexSpace | ByBreakpoint<NumericSpace>) => {
+    if (typeof space === 'number' || typeof space === 'string') {
+        return {
+            [styles.vars.space]: calcSpaceValue(space),
+        };
+    }
+    const vars = {
+        [styles.vars.spaceMobile]: calcSpaceValue(space.mobile),
+        [styles.vars.spaceDesktop]: calcSpaceValue(space.desktop),
+    };
+    if (space.tablet) {
+        vars[styles.vars.spaceTablet] = calcSpaceValue(space.tablet);
+    }
+    return vars;
+};
 
 type Props = {
-    space:
-        | -16
-        | -12
-        | -8
-        | -4
-        | -2
-        | 0
-        | 2
-        | 4
-        | 8
-        | 12
-        | 16
-        | 24
-        | 32
-        | 40
-        | 48
-        | 56
-        | 64
-        | 'between'
-        | 'around'
-        | 'evenly';
+    space: FlexSpace | ByBreakpoint<NumericSpace>;
```

Comment:
support setting `space` by breakpoint as we already support in `Stack` and `Box` components